### ISSUE TITLE
`local-devices` mislabeled as code-injection

### ISSUE THREAD

**Issue by ericcornelissen** [cristianstaicu/SecBench.js#15]
The vulnerability in [`local-devices@2.0.0`](https://github.com/cristianstaicu/SecBench.js/tree/efc9da1acc50bceddd40f3ce48d5f0f426feb614/code-injection/local-devices_2.0.0) is mislabeled as code injection while it is actually command injection. The [listed code location](https://github.com/cristianstaicu/SecBench.js/blob/efc9da1acc50bceddd40f3ce48d5f0f426feb614/sink_locations_ace_breakout.txt#L29) points to an call to `child_process.exec` in ([code on npm](https://www.npmjs.com/package/local-devices/v/2.0.0?activeTab=code)):

```js
/**
 * Reads the arp table for a single address.
 */
function arpOne (address) {
  return cp.exec('arp -n ' + address).then(parseOne)
}
```

Also, the [Proof of Concept](https://github.com/cristianstaicu/SecBench.js/blob/efc9da1acc50bceddd40f3ce48d5f0f426feb614/code-injection/local-devices_2.0.0/local-devices.test.js#L8) has a command injection payload and not a code injection payload.

**Comment by cristianstaicu** [comment]
I agree with you, however Snyk labeled this vulnerability as code injection at the time (https://security.snyk.io/vuln/SNYK-JS-LOCALDEVICES-459898), so we decided to keep this label. Did you find any other such entry in the dataset? I am considering moving it, as you suggested, into the command-injection folder.

**Comment by ericcornelissen** [comment]
> I agree with you, however Snyk labeled this vulnerability as code injection at the time ([security.snyk.io/vuln/SNYK-JS-LOCALDEVICES-459898](https://security.snyk.io/vuln/SNYK-JS-LOCALDEVICES-459898)), so we decided to keep this label.

I see, though their own description disagrees:

> > The package does not validate input on ip addresses and concatenates it to an **exec call**, allowing attackers to run **arbitrary commands** in the system.

and the GHA entry does label it as command injection, see https://github.com/advisories/GHSA-w725-67p7-xv22


> Did you find any other such entry in the dataset?

So far not. I will probably only be evaluating the [code injections](https://github.com/cristianstaicu/SecBench.js/tree/bc315621913899dcaa7613cd60948514ae63bdfe/code-injection) and have reached (in the GitHub directory order) up to and including `mosc_1.0.0`.

**Comment by ericcornelissen** [comment]
Another case: [`open`](https://github.com/cristianstaicu/SecBench.js/tree/bc315621913899dcaa7613cd60948514ae63bdfe/code-injection/open_0.0.5), see:

https://github.com/cristianstaicu/SecBench.js/blob/bc315621913899dcaa7613cd60948514ae63bdfe/code-injection/open_0.0.5/open.test.js#L14

----

- Snyk (misleading): https://security.snyk.io/vuln/SNYK-JS-OPEN-174041
- GHSA (correct): https://github.com/advisories/GHSA-28xh-wpgr-7fm8
- Sink location source: https://github.com/cristianstaicu/SecBench.js/blob/bc315621913899dcaa7613cd60948514ae63bdfe/test_sink_location.txt#L36
- Sink location: https://github.com/pwnall/node-open/blob/2d20d7c8946b2da243a59d9c727953bf06d77611/lib/open.js#L58

**Comment by ericcornelissen** [comment]
Another case: [`wifiscanner`](https://github.com/cristianstaicu/SecBench.js/tree/bc315621913899dcaa7613cd60948514ae63bdfe/code-injection/wifiscanner_1.0.1), see:

https://github.com/cristianstaicu/SecBench.js/blob/bc315621913899dcaa7613cd60948514ae63bdfe/code-injection/wifiscanner_1.0.1/wifiscanner.test.js#L9

---

- Snyk (misleading): https://security.snyk.io/vuln/SNYK-JS-WIFISCANNER-574786
- GHSA (mostly correct): https://github.com/advisories/GHSA-m6rw-m2v9-7hx4
- Sink location source: https://github.com/cristianstaicu/SecBench.js/blob/bc315621913899dcaa7613cd60948514ae63bdfe/test_sink_location.txt#L30-L30
- Sink location: https://github.com/thingsSDK/wifiscanner/blob/9695ed53fba4e1b7a56a918c009cba07ead8a5a4/scanners/wifiscanner.js#L12

**Comment by ericcornelissen** [comment]
That's all entries in [code-injection](https://github.com/cristianstaicu/SecBench.js/tree/bc315621913899dcaa7613cd60948514ae63bdfe/code-injection) that are mislabeled from what I can tell.

#### Overview

- [`local-devices`](https://github.com/cristianstaicu/SecBench.js/issues/15#issue-2556278048)
- [`open`](https://github.com/cristianstaicu/SecBench.js/issues/15#issuecomment-2394466242)
- [`wifiscanner`](https://github.com/cristianstaicu/SecBench.js/issues/15#issuecomment-2394526565)
